### PR TITLE
Fixed overflow issues with int64 fill value handling

### PR DESCRIPTION
**Pull request by srherbener**
## Description

This PR fixes ctests that exhibited overflow errors when handling int64 fill values. The typical way this shows up is when converting the date time values in the time offset format (float32) into date time values in the epoch format (int64).

This involves an update of test reference files for three tests:

- test_iodaconv_prepbufr_ncep_api_adpsfc2ioda
- test_iodaconv_prepbufr_ncep_api_sfcshp2ioda
- test_iodaconv_bufr_ncep_prepbufr_adpupa_api

I verified that the only thing that changed in the test reference files was the values of the fill values used for dateTime and releaseTime variables (which are stored in the epoch format in int64 data type). Also verified that the fill value spec in the test reference files for dateTime and releaseTime are consistent with the actual values in the variable data (so the ioda reader will be able to recognize these as missing).

## Issue(s) addressed

Partially addresses #1378

## Dependencies

List the other PRs that this PR is dependent on:
None

## Impact

Expected impact on downstream repositories:
The only impact is for cases where the actual value of missing values in dateTime  and releaseTime matter. The expectation is that this means no impact.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation (code comments)
- [x] I have run the unit tests before creating the PR
- [x] Test on HPC platform - Orion, GNU, OpenMPI

Note that differences between Orion and the Mac M1 were what triggered looking into this, so it would be good to verify that both of these get the same results.

### CODE REVIEW

File: test/testinput/bufr_ncep_prepbufr_adpupa.py [hunk 1]
```diff
@@ -67,10 +68,19 @@ def test_bufr_to_ioda(DATA_PATH, OUTPUT_PATH, date):
...
     # Time variable
+    # The time is entering in as a float32 value representing an offset from the cycle
+    # time in hours (so fractions of hours can exist). This needs to be first converted
+    # to seconds while still a float32, then converted to an int64 for the dateTime variable.
+    # Another consideration is that the get function returns a masked array with an
+    # appropriate fill value assigned. When converting to an int64, the fill value needs
+    # to get updated, and then before writing into the output ioda file, the masked array
+    # function filled() needs to be called which will convert the values marked invalid
+    # to the fill value.
     hrdr = r.get('timeOffset')
```
Comment (rmclaren) [inline test/testinput/bufr_ncep_prepbufr_adpupa.py:79]: delete this line

**Review by rmclaren — APPROVED** [review]
@srherbener Thanks so much for fixing these!

File: test/testinput/bufr_ncep_prepbufr_adpupa.py (hunk 1 above)
Comment (srherbener) [reply test/testinput/bufr_ncep_prepbufr_adpupa.py]: Thanks for catching this. I've removed the line.

**Review by nicholasesposito — APPROVED** [review]
Looks fine as a fix to the v1 scripts. For the record, there are v2 scripts that I don't believe have the same issue. They're in branches on GDASApp but I don't know if there is a plan to publish them at the moment.

**Review by PatNichols — APPROVED** [review]
Thanks @srherbener !

**Pull request merged**